### DISCUSSION TITLE
Can Emotion's css tool be used in conjunction with string classNames in Reakit?

### DISCUSSION THREAD

**Discussion by Nantris** [ariakit/ariakit#452] [Questions] [answered]
Taking this example from the docs:

```js
import { Button } from "reakit";
import { css } from "emotion";

function Example() {
  return (
    <Button
      className={css({
        color: "white",
        background: "red",
        "&:not([disabled]):hover": { background: "green" },
        "&:not([disabled]):active": { background: "blue" }
      })}
    >
      Button
    </Button>
  );
}
```

Is there some way we can also use string `classNames` in conjunction? Say I had a class called `primaryButton` or something like that that I wanted to use in addition to the `css` shown above. Is that possible?

 I think what I'm asking is not something that conforms with best practices, but it would make it easier to move from SCSS styling to Reakit in a piecemeal fashion.

[a comment by an automated account is not shown]

**Comment by diegohaz** [comment]
There's nothing special on Reakit regarding this. So I guess you'll have better answers asking on `emotion` channels.

As far as I know, `css` returns a string, and you can concatenate it with other strings. There's also a `cx` helper:
```jsx
import { Button } from "reakit";
import { css, cx } from "emotion";

function Example() {
  const className = css({
    color: "white",
    background: "red",
    "&:not([disabled]):hover": { background: "green" },
    "&:not([disabled]):active": { background: "blue" }
  });
  return (
    // <Button className={cx(className, "a")}>
    // or
    <Button className={`${className} a`}>
      Button
    </Button>
  );
}
```

**Comment by Nantris** [comment]
@diegohaz specifically the reason I asked is because with `emotion` you pass `css` as a prop called `css` which leaves the `className` prop open for use as usual.  That's not the case with `reakit` though.

For example, the [emotion Array example](https://emotion.sh/docs/object-styles):

```js
/** @jsx jsx */
import { jsx } from '@emotion/core'

render(
  <div
    css={[
      { color: 'darkorchid' },
      { backgroundColor: 'hotpink' },
      { padding: 8 }
    ]}
  >
    This is darkorchid with a hotpink background and 8px of
    padding.
  </div>
)
```

Your example is very helpful though. I think what you're saying is that it returns a `className` on the fly, and that makes a lot of sense. I'll give that a shot! Thanks for taking the time @diegohaz .

**Comment by diegohaz** [comment]
Emotion's css prop should work with Reakit. Isn't it working?

**Comment by Nantris** [comment, marked as answer]
@diegohaz It looks like the reason I wasn't seeing changes applied is [this bug that I think is in Emotion](https://github.com/emotion-js/emotion/issues/1524). Doesn't appear to be anything to do on `reakit`'s side.

That said, I didn't have any luck at all with this pattern in my attempts to get this working:

```js

import { Box, Button } from 'reakit';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

const StyledButton = styled(Button)``

const FinalButton = props => (
  <StyledButton
    {...{
      ...props,
      css: css`
        color: red;
      `,
    }}>
    Some Text
  </StyledButton>)
```
